dev/benchmarks/macrobenchmarks/lib/src/web/gallery/gallery_recorder.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'package:gallery/benchmarks/gallery_automator.dart';
+
+import 'package:macrobenchmarks/src/web/recorder.dart';
+
+/// A recorder that measures frame building durations for the Gallery.
+class GalleryRecorder extends WidgetRecorder {
+  GalleryRecorder({
+    @required this.benchmarkName,
+    @required this.reporter,
+    @required this.reportError,
+    @required this.runCriterion,
+    this.testScrollsOnly = false,
+  }) : super(name: benchmarkName, warmUpFrameCount: 5);
```

Comment:
Is this  number (5) similar to other tests that have recorders?

Reply:
The other tests with recorders use the default warm-up frame count of 100 (the first 100 frames are ignored).

From what I understand, the other tests mostly contain animations — for example, [bench_card_infinite_scroll](https://github.com/flutter/flutter/blob/master/dev/benchmarks/macrobenchmarks/lib/src/web/bench_card_infinite_scroll.dart) is an infinite scrolling test. The frames are very similar, so the first 100 frames are ignored in a total of 300 frames.

The Gallery test opens up different studies and widget demos, and there isn't any animation at the beginning, so the actual warm-up frames should be much fewer than 100.

Reply:
I'm not sure 5 frames will be enough to warm-up the JIT. I'm wondering if it makes sense to control the warm-up based on higher-level concepts, such as "the number of demos opened", just like you decide when the benchmark is `_finished` based on demos exercised.

Reply:
@yjbanov My response is below at https://github.com/flutter/flutter/pull/57576#issuecomment-633480886